### PR TITLE
[Feature] Add rectangle and triangle sdf_func

### PR DESCRIPTION
**Pull request by AndPuQing**
### PR types
New features

### PR changes
APIs

### Describe

In this pull request, I have added the sdf_func implementations for rectangles and triangles. The visual result can be tested using the following code:

```python
import ppsci
import numpy as np

from matplotlib import pyplot as plt

geo = ppsci.geometry.Triangle((0, 0), (1, 0), (0, 1))

x = np.linspace(-0.5, 1.3, 1000)
y = np.linspace(-0.5, 1.3, 1000)
X, Y = np.meshgrid(x, y)
x = X.flatten()
y = Y.flatten()
Z = geo.sdf_func(np.stack([x, y], axis=-1)).reshape(X.shape)
plt.contour(X, Y, Z, 0, colors="black")
plt.contourf(X, Y, Z, 20, cmap="RdBu_r")
plt.colorbar()
plt.show()
```

**Visual Result** 

output
output1

https://github.com/PaddlePaddle/PaddleScience/issues/376

### CODE REVIEW

File: ppsci/geometry/geometry_2d.py [hunk 1]
```diff
@@ -177,6 +177,29 @@ def is_valid(vertices):
...
+        Returns:
+            np.ndarray: Unsquared SDF values of input points, the shape is `[N, 1]`.
+
+        NOTE: This function usually returns ndarray with negative values, because
+        according to the definition of SDF, the SDF value of the coordinate point inside
+        the object(interior points) is negative, the outside is positive, and the edge
+        is 0. Therefore, when used for weighting, a negative sign is often added before
+        the result of this function.
+        """
+        assert points.shape[1] == 2, "The shape of points must be [N, 2]."
```
Comment (HydrogenSulfate) [inline ppsci/geometry/geometry_2d.py:196]: 1. 入参检查用 `if ...: raise ValueError`，不要用 assert
2. `"The shape of points must be [N, 2]."` -> `"Shape of given points should be [*, 2], but go t {points.shape}"`

File: ppsci/geometry/geometry_2d.py [hunk 2]
```diff
@@ -349,6 +372,49 @@ def random_boundary_points(self, n, random="pseudo"):
...
```
Comment (HydrogenSulfate) [inline ppsci/geometry/geometry_2d.py:391]: 同上

File: ppsci/geometry/geometry_2d.py [hunk 3]
```diff
@@ -349,6 +372,49 @@ def random_boundary_points(self, n, random="pseudo"):
...
+            * np.clip(np.dot(v2p, self.v23.reshape(2, -1)) / self.l23**2, 0, 1)
+            - v2p
+        )
+        pq3 = (
+            self.v31
+            * np.clip(np.dot(v3p, self.v31.reshape(2, -1)) / self.l31**2, 0, 1)
+            - v3p
+        )
+
+        s = self.is_inside(points).reshape(-1, 1) * 2 - 1
```
Comment (HydrogenSulfate) [inline ppsci/geometry/geometry_2d.py:411]: 变量名字建议不要单个小写字母，命名尽量能表明变量本身的意义

File: ppsci/geometry/geometry_2d.py [hunk 4]
```diff
@@ -177,6 +177,29 @@ def is_valid(vertices):
...
+
+        NOTE: This function usually returns ndarray with negative values, because
+        according to the definition of SDF, the SDF value of the coordinate point inside
+        the object(interior points) is negative, the outside is positive, and the edge
+        is 0. Therefore, when used for weighting, a negative sign is often added before
+        the result of this function.
+        """
+        assert points.shape[1] == 2, "The shape of points must be [N, 2]."
+        center = (self.xmin + self.xmax) / 2
+        q = np.abs(points - center) - np.array([self.xmax - self.xmin]) / 2
```
Comment (HydrogenSulfate) [inline ppsci/geometry/geometry_2d.py:198]: 变量名字建议不要单个小写字母，命名尽量能表明变量本身的意义

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: ppsci/geometry/geometry_2d.py (hunk 1 above)
Comment (AndPuQing) [reply ppsci/geometry/geometry_2d.py]: done

**Review by HydrogenSulfate — APPROVED** [review]
LGTM

**Pull request merged**